plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -80,13 +77,30 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
 
     fun registerForVerification(declaration: FirSimpleFunction) {
         val signature = embedFullSignature(declaration.symbol)
+        val returnTarget = returnTargetProducer.getFresh(signature.callableType.returnType)
+        val stmtCtx = signature.createStatementContext(returnTarget, scopeIndexProducer.getFresh())
+
         // Note: it is important that `body` is only set after `embedUserFunction` is complete, as we need to
         // place the embedding in the map before processing the body.
         embedUserFunction(declaration.symbol, signature).apply {
-            body = convertMethodWithBody(declaration, signature)
+            body = stmtCtx.convertMethodWithBody(declaration, signature, returnTarget)
         }
     }
 
+    private fun FullNamedFunctionSignature.createStatementContext(
```

Comment:
Should this really be an extension function?  It seems a bit arbitrary to make it one.